### DISCUSSION TITLE
Keeping some collections on data layer (or self hosted database) only

### DISCUSSION THREAD

**Discussion by leomp12** [tinacms/tinacms#4081] [Help] [answered]
In production deployment I would like to read some single file collections from local filesystem (sync with GitHub) for zero latency (`TINA_PUBLIC_IS_LOCAL` ? 🤔 ) and other collections (e.g. blog posts) from data layer only, never pushed to GitHub.
Is it possible? Would it be complicated? Self-hosting would be required?

**Comment by kldavis4** [comment]
@leomp12 I'm not totally following, but can you clarify when you will be querying these data layer only collections? Is this at build time? In theory you can write anything into the self-hosted database, but there is nothing builtin to Tina that would do this for your.

**Reply by leomp12** [reply]
I'll be querying these data layer only collections on SSR function with specific routes.

To be clearer, it is an ecommerce with commerce data coming from external APIs, some (single-file) collections will define styles, sections, site config... Are required on all routes and also used at build (bundling) time.
Other collections (blog posts, extra pages) are not that important (those routes can be bit slower) and may become big enough to be bad to commit on repository, I also don't want them to trigger repository CI/CD.

**Reply by kldavis4** [reply]
So for these collections that aren't committed to the repository, how would they be populated?

**Reply by leomp12** [reply]
They would be populated on CMS and stored on database only.
It's kind of easy to do this with some other git-based CMS with a proxy to GitHub API and custom middlewares for some routes (matching the collection name), but I have no idea yet how this works on Tina 🙃

**Reply by leomp12** [reply]
Becomes a non-100%-git-based CMS 😄

**Reply by kldavis4** [reply]
What specifically do you mean by "populated on CMS" ?

**Reply by leomp12** [reply]
Admin user populate this data on CMS, no way to edit it directly on repository source.
When saved, data is stored directly on database, never pushed no repository. 
CMS reads this data from database directly even if it "doesn't know about it" (e.g. proxying GitHub API with custom middleware to read the data directly on database for some filepaths).

**Reply by kldavis4** [reply, marked as answer]
Thanks for the clarification. With self-hosted, you are responsible for managing the puts / deletes to the git provider, so you could theoretically control what collection paths are committed or not. I think the only issue with this, at the moment, is that TinaCMS treats the data layer as ephemeral, so when the schema changes, the subset of the db used for all collections is wiped and recreated. Anything in these collections that isn't committed to the database would be lost. The basic idea being your git repository is the ultimate source of truth for the data layer. Theoretically, we could have a property on collections which marks them as non-ephemeral and puts them into a different namespace so they don't get wiped. We'd probably still need to keep the indexes for this content in the ephemeral db so that those are updated when the schema changes, and there are probably other things that might complicate this. 

So I think the answer is, it may be possible with modifications, but in the current version of tina, we don't support data layer only collections.
